### PR TITLE
chore: setup helper-plugin for TS conversion

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

Building upon #17398 and using the helper-plugin as a starting point we were able to define a comprehensive TS eslint config (strapi/eslint-config#6). It adds the necessary changes for jest to parse ts / tsx files, some extra eslint bits unique to the CMS have been added to the local version

### Why is it needed?

So we have a strong eslint config with recommended standards and explanations.

### Related issue(s)/PR(s)

* requires strapi/eslint-config#6 be merged first and we update the dependency here.
* requires #17486
* resolves CONTENT-1642

### CODE REVIEW

File: packages/core/helper-plugin/.eslintrc.js [hunk 1]
```diff
@@ -1,4 +1,31 @@
 module.exports = {
   root: true,
-  extends: ['custom/front'],
+  extends: ['custom/front/typescript'],
+  parserOptions: {
+    project: ['./tsconfig.eslint.json'],
+  },
+  settings: {
+    'import/resolver': {
```
Comment (alexandrebodin) [inline packages/core/helper-plugin/.eslintrc.js:8]: Surprised this isn't picked up via the parserOptions already

File: packages/core/helper-plugin/package.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
     "test:front:cov": "run -T jest --config ./jest.config.front.js --coverage",
     "test:front:watch": "run -T jest --config ./jest.config.front.js --watchAll",
     "watch": "cross-env NODE_ENV=development webpack-cli -w",
-    "lint": "run -T eslint ."
+    "lint": "run -T eslint . --ext .js,.jsx,.tsx,.ts"
```
Comment (alexandrebodin) [inline packages/core/helper-plugin/package.json:44]: can be configured in the eslint settings directly I believe

**Review by alexandrebodin — COMMENTED** [review]
Should I expect the changes from js to jsx to becomes tsx in this PR ?

File: packages/core/helper-plugin/.eslintrc.js (hunk 1 above)
Comment (joshuaellis) [reply packages/core/helper-plugin/.eslintrc.js:8]: yeah i was following the set up instructions from here – https://github.com/import-js/eslint-import-resolver-typescript

File: packages/core/helper-plugin/package.json (hunk 2 above)
Comment (joshuaellis) [reply packages/core/helper-plugin/package.json:44]: Yeah I couldn't find this one out for the global setting, do you know which one it is? :(

**Comment by joshuaellis** [comment]
> Should I expect the changes from js to jsx to become tsx in this PR ?

No, there's a lot of files to be converted and I think if we get this package to handle TS which is now can do, I can probably do some follow up PRs to convert pieces, not a lot of this package is tested but it's incredibly foundational so i'd rather go a little bit slower to be safer.

File: packages/core/helper-plugin/package.json (hunk 2 above)
Comment (gu-stav) [reply packages/core/helper-plugin/package.json:44]: I don't think that is possible either :)

> This request came up multiple times before. Unfortunately, as @mysticatea explained, it's not really possible to do, especially, when you add complexity of shareable configs and cascading into the picture. So I'm going to close this issue, sine we can't address this.

https://github.com/eslint/eslint/issues/11223

**Review by gu-stav — APPROVED** [review]
Changes look good to me - well done 👏🏼 

@joshuaellis Should we setup a tracking issue again similar to what we did in the design-system? I'd find that really helpful to keep track of the progress, distribute the work and make sure we migrate the things first that have more value to us in the coming months.

If you want I could take care of this ...

File: packages/core/helper-plugin/package.json (hunk 2 above)
Comment (joshuaellis) [reply packages/core/helper-plugin/package.json:44]: yeah it won't pick up `.jsx` files for some reason 🤔  it'll do `.ts`, `.tsx` & `.js` but not `.jsx`....

**Comment by joshuaellis** [comment]
> Changes look good to me - well done 👏🏼
> 
> @joshuaellis Should we setup a tracking issue again similar to what we did in the design-system? I'd find that really helpful to keep track of the progress, distribute the work and make sure we migrate the things first that have more value to us in the coming months.
> 
> If you want I could take care of this ...

hmmmmm, im a bit hesitant towards this, I don't really think we should hand this out to the community, there's a lot of undocumented confusing logic here & with an in-depth knowledge of the CMS product we can probably identify pieces we either don't need or should be moved and highlight them as we go.

In addition we need this converted to move forward with the rest of the CMS migration sooooo really it all needs doing (to "_some_" level), I think realistically we just have to make an Epic and chunk it into sections as tasks and deliver it 🤷🏼‍♀️ I don't mind starting this in the next sprint as we're refining the bigger tasks. From there though, we can prioritise the most "valuable" parts of the codebase...

**Comment by gu-stav** [comment]
> I think realistically we just have to make an Epic and chunk it into sections as tasks and deliver it 🤷🏼‍♀️ I don't mind starting this in the next sprint as we're refining the bigger tasks. From there though, we can prioritise the most "valuable" parts of the codebase...

Makes sense - we can also keep the tracking internally, as long as we have some kind of overview where we are at.

The problem I see with an EPIC is that epics are tied to one board, whereas the TS rework should be cross-squad. Maybe it would be good if we prepare a plan and then see which parts can be done by whom and involve Paridhi, because it might also depend on what e.g. studio will be working.

**Comment by joshuaellis** [comment]
> whereas the TS rework should be cross-squad.

Sure, i don't anything is defined for this at the moment i.e. who's working on what and where, i think we'd get more answers discussing this in Slack to be honest

File: packages/core/helper-plugin/package.json (hunk 2 above)
Comment (alexandrebodin) [reply packages/core/helper-plugin/package.json:44]: ho ok that's strange. I usually just set this up https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/extensions.md

**Comment by alexandrebodin** [comment]
> > whereas the TS rework should be cross-squad.
> 
> Sure, i don't anything is defined for this at the moment i.e. who's working on what and where, i think we'd get more answers discussing this in Slack to be honest

We can you the v5 Board for this

File: packages/core/helper-plugin/.eslintrc.js (hunk 1 above)
Comment (alexandrebodin) [reply packages/core/helper-plugin/.eslintrc.js:8]: ok, we never needed it for the other ts package I'd be curious to know why someday :D

**Review by gu-stav — APPROVED** [review]

**Pull request merged**